### ISSUE TITLE
colmap2adop crashes on assertion: info.ExposureBiasValue == 0

### ISSUE THREAD

**Issue by trzy** [darglein/ADOP#46]
Hi,

I'm trying to learn how to use ADOP for my own set of images. I've never really used colmap before but using audomatic reconstruction in the GUI, was able to produce a dense and sparse reconstruction. Note that my .ply point cloud file is located in $SCENE_BASE/dense/0 rather than $SCENE_BASE as in the usage example in README.md. 

I get tripped up by some assertion and I'm not sure if there is a workaround. Here is my output:

```
bart@Technodrome:~/projects/ADOP$ build/bin/colmap2adop --sparse_dir $SCENE_BASE/sparse/0 --image_dir scenes/lakeridge_pool/images --point_cloud_file $SCENE_BASE/dense/0/fused.ply --output_path scenes/lakeridge_pool --scale_intrinsics 1 --render_scale 1
register neural render info
Preprocessing Colmap scene /home/bart/projects/lakeridge/sparse/0 -> scenes/lakeridge_pool
Num cameras 1
id: 0 camera model: 2  K: 3088.08 3088.08    2016    1512       0 Dis: 0.0116571         0         0         0         0         0         0         0
Num images 186
IMG_3136.JPG 4 -> 0 0  Position:  -1.89089 -0.234411   4.52033
IMG_3137.JPG 3 -> 1 0  Position:  0.378659 -0.451968   4.87581
IMG_3138.JPG 2 -> 2 0  Position: 0.564876 0.873621  4.80398
IMG_3139.JPG 1 -> 3 0  Position: 1.16454  1.5363 4.52841
IMG_3140.JPG 6 -> 4 0  Position:   1.55933 -0.558918   4.64452
IMG_3141.JPG 7 -> 5 0  Position:   3.61732 -0.505346   3.36747
IMG_3142.JPG 9 -> 6 0  Position:     4.8889 -0.0918832   0.448115
IMG_3143.JPG 5 -> 7 0  Position:  4.18467 0.241358 -2.47252
IMG_3144.JPG 8 -> 8 0  Position:  2.62575 0.569915 -3.71823
IMG_3145.JPG 13 -> 9 0  Position:   2.57384 -0.831334  -3.69662
IMG_3146.JPG 17 -> 10 0  Position:   2.10184 -0.460205  -3.96549
IMG_3147.JPG 11 -> 11 0  Position:  2.1416 1.04326 -3.8367
IMG_3148.JPG 12 -> 12 0  Position: 0.958832  1.39273 -4.15396
IMG_3149.JPG 10 -> 13 0  Position: -1.41954  1.28613 -4.09003
IMG_3150.JPG 18 -> 14 0  Position:  -1.41911 -0.200994  -4.27028
IMG_3151.JPG 32 -> 15 0  Position: -1.43549 -1.36029 -4.03637
IMG_3152.JPG 14 -> 16 0  Position: -3.59765 0.939916 -2.84264
IMG_3153.JPG 15 -> 17 0  Position:  -4.34177 0.0643341  -2.28495
IMG_3154.JPG 16 -> 18 0  Position: -4.42102 0.787602  -1.9721
IMG_3155.JPG 23 -> 19 0  Position: -4.39101  1.30871 -1.75209
IMG_3156.JPG 31 -> 20 0  Position:  -4.43574 -0.345473  -2.06734
IMG_3157.JPG 19 -> 21 0  Position:  -4.68322 -0.326334    2.6192
IMG_3158.JPG 22 -> 22 0  Position: -2.70991 -1.00033  4.86697
IMG_3159.JPG 24 -> 23 0  Position: -0.464814  -1.40312   5.64586
IMG_3160.JPG 20 -> 24 0  Position: -0.494801  0.529923     5.788
IMG_3161.JPG 25 -> 25 0  Position:   2.23685 -0.164267    5.4254
IMG_3162.JPG 26 -> 26 0  Position: 0.75389 -2.9956  4.9654
IMG_3163.JPG 21 -> 27 0  Position: -1.58557 -3.04788  4.72871
IMG_3164.JPG 27 -> 28 0  Position:    1.2923 -0.726255   5.60195
IMG_3165.JPG 29 -> 29 0  Position:  0.175207 -0.805053   4.98503
IMG_3166.JPG 28 -> 30 0  Position:  0.196862 -0.519683   4.44226
IMG_3167.JPG 30 -> 31 0  Position: 0.0440985 -0.300632   4.44549
IMG_3168.JPG 38 -> 32 0  Position: -1.78238 0.238353  4.06945
IMG_3169.JPG 34 -> 33 0  Position:  -3.69564 -0.224309   2.50958
IMG_3170.JPG 33 -> 34 0  Position:   -4.4071 -0.138796  0.465772
IMG_3171.JPG 37 -> 35 0  Position: -3.78188 0.115589 -2.40882
IMG_3172.JPG 39 -> 36 0  Position: -3.00593 0.199341 -3.37477
IMG_3174.JPG 36 -> 37 0  Position:  3.94743 0.290594 -2.36559
IMG_3175.JPG 51 -> 38 0  Position:    4.51454 -0.0414131   0.474309
IMG_3176.JPG 50 -> 39 0  Position:   3.83682 -0.215717   2.33619
IMG_3177.JPG 41 -> 40 0  Position:   1.98051 -0.429242   3.94924
IMG_3178.JPG 40 -> 41 0  Position:  -2.96871 -0.633854     3.862
IMG_3179.JPG 46 -> 42 0  Position:  -2.08799 -0.930789   4.06483
IMG_3180.JPG 43 -> 43 0  Position: -2.26218  0.23649  4.02214
IMG_3181.JPG 48 -> 44 0  Position:  -2.81931 -0.922176   2.96086
IMG_3182.JPG 44 -> 45 0  Position: -2.9285 -0.3508 2.18193
IMG_3183.JPG 42 -> 46 0  Position:  -3.69997 0.0438288  0.069077
IMG_3184.JPG 47 -> 47 0  Position:  -2.31767 -0.330573   1.94316
IMG_3185.JPG 49 -> 48 0  Position:   -3.05043 -0.0346761   0.242605
IMG_3186.JPG 45 -> 49 0  Position:  -2.99199 0.0809456 -0.701316
IMG_3187.JPG 55 -> 50 0  Position:  -1.43001 -0.253776   1.97616
IMG_3188.JPG 53 -> 51 0  Position:   -2.33605 -0.0550523   0.811886
IMG_3189.JPG 52 -> 52 0  Position:  -2.49588 0.0490388 -0.164555
IMG_3190.JPG 57 -> 53 0  Position: -2.26337 0.328444 0.917239
IMG_3191.JPG 56 -> 54 0  Position: -0.957613  0.458248   2.18509
IMG_3192.JPG 54 -> 55 0  Position:   0.5938 0.158907  2.33484
IMG_3193.JPG 59 -> 56 0  Position: -0.144778 -0.327327    1.8413
IMG_3194.JPG 60 -> 57 0  Position:   -1.1909 -0.174417   1.45126
IMG_3195.JPG 58 -> 58 0  Position:   -1.82146 -0.0557306   0.614374
IMG_3196.JPG 61 -> 59 0  Position: 0.607432 -0.40251  1.86614
IMG_3197.JPG 63 -> 60 0  Position: 0.568675 0.173578  1.91622
IMG_3198.JPG 62 -> 61 0  Position:  -0.681502 -0.0274086    1.87494
IMG_3199.JPG 65 -> 62 0  Position:  -1.32193 -0.513084   1.44523
IMG_3200.JPG 68 -> 63 0  Position:  -1.74884 -0.233343   1.00627
IMG_3201.JPG 64 -> 64 0  Position:  1.86615 -0.17015 0.804502
IMG_3202.JPG 67 -> 65 0  Position:    2.02134 0.00520135  -0.556621
IMG_3203.JPG 66 -> 66 0  Position:   1.66071 -0.422608   1.70771
IMG_3204.JPG 70 -> 67 0  Position: -0.0469477   -0.34337    2.35263
IMG_3205.JPG 71 -> 68 0  Position:  -1.29148 -0.222989   2.00191
IMG_3206.JPG 74 -> 69 0  Position: -0.0463777   0.565541    2.32211
IMG_3207.JPG 75 -> 70 0  Position:  1.38704 0.369775  1.94309
IMG_3208.JPG 69 -> 71 0  Position:  2.40951 0.127066 0.556516
IMG_3209.JPG 73 -> 72 0  Position:   2.58436 -0.545839    1.7431
IMG_3210.JPG 76 -> 73 0  Position:  2.99488 0.231932  -1.2595
IMG_3211.JPG 72 -> 74 0  Position:  2.37329 -1.04808  1.80903
IMG_3212.JPG 81 -> 75 0  Position: 2.24534 0.17835 2.21718
IMG_3213.JPG 79 -> 76 0  Position: 0.198279 0.165949  3.10946
IMG_3214.JPG 77 -> 77 0  Position: -0.820161 -0.694361   2.92618
IMG_3215.JPG 78 -> 78 0  Position:   2.04264 -0.991337   2.92062
IMG_3216.JPG 84 -> 79 0  Position:   1.90179 -0.872609   3.38771
IMG_3217.JPG 82 -> 80 0  Position: 1.58835 0.27918 3.63485
IMG_3218.JPG 80 -> 81 0  Position: -0.248228 0.0619577   3.95284
IMG_3219.JPG 91 -> 82 0  Position:  -1.12198 -0.701833   3.73063
IMG_3220.JPG 83 -> 83 0  Position:   3.45532 -0.377205   2.01011
IMG_3221.JPG 87 -> 84 0  Position:    4.01337 -0.0581364 -0.0796334
IMG_3222.JPG 86 -> 85 0  Position:   3.43061 0.0834168  -2.13259
IMG_3223.JPG 88 -> 86 0  Position:  2.00401 0.306856 -3.51632
IMG_3224.JPG 90 -> 87 0  Position: -0.804239  0.594517  -3.76472
IMG_3225.JPG 85 -> 88 0  Position: -0.981489  -1.12208  -3.57139
IMG_3226.JPG 89 -> 89 0  Position:  -2.60022 -0.455816  -2.81094
IMG_3227.JPG 92 -> 90 0  Position: -2.88278  0.45049 -2.52193
IMG_3228.JPG 99 -> 91 0  Position:   -3.7234  0.289773 -0.708272
IMG_3229.JPG 102 -> 92 0  Position:  -3.39322 -0.326944   1.69969
IMG_3230.JPG 104 -> 93 0  Position:   -2.3301 -0.304962   2.97787
IMG_3231.JPG 93 -> 94 0  Position:  -0.46139 -0.942099   3.84172
IMG_3232.JPG 94 -> 95 0  Position: -0.737662  -1.00345   3.63189
IMG_3233.JPG 96 -> 96 0  Position: -0.557072  0.502447   3.76734
IMG_3234.JPG 106 -> 97 0  Position: -2.90301 -0.56731  2.45789
IMG_3235.JPG 98 -> 98 0  Position: -0.579748  -1.42758   3.29721
IMG_3236.JPG 95 -> 99 0  Position: -0.577174 -0.379586   3.54844
IMG_3237.JPG 97 -> 100 0  Position: -0.0801768    -1.9624    2.95499
IMG_3238.JPG 100 -> 101 0  Position: -0.0536503  -0.911851    3.43921
IMG_3239.JPG 103 -> 102 0  Position: 0.306203   -1.803  3.03705
IMG_3240.JPG 101 -> 103 0  Position:  0.315391 -0.353755   3.51598
IMG_3241.JPG 105 -> 104 0  Position: -1.51136 -0.26495  3.19468
IMG_3242.JPG 110 -> 105 0  Position:   -3.32458 -0.0798645     1.3056
IMG_3243.JPG 114 -> 106 0  Position:  -3.59386 0.0584838 -0.273204
IMG_3244.JPG 107 -> 107 0  Position:   1.58976 -0.519914   3.11205
IMG_3245.JPG 109 -> 108 0  Position:  1.81496 0.630129   2.9771
IMG_3246.JPG 125 -> 109 0  Position:   3.58685 -0.213295  0.248723
IMG_3247.JPG 108 -> 110 0  Position:  0.578744 -0.702924   3.54497
IMG_3248.JPG 111 -> 111 0  Position: -1.51073 -0.50643  3.28996
IMG_3249.JPG 161 -> 112 0  Position:  -3.03582 -0.338988   2.03637
IMG_3250.JPG 119 -> 113 0  Position: -0.0891438   -0.95233    3.84892
IMG_3251.JPG 120 -> 114 0  Position:  -2.42481 -0.551758   3.07371
IMG_3252.JPG 115 -> 115 0  Position: -3.44133 -0.26377  1.96637
IMG_3253.JPG 113 -> 116 0  Position:  -3.81353 0.0840584  0.114018
IMG_3254.JPG 121 -> 117 0  Position:  -3.80079  0.190982 -0.012798
IMG_3255.JPG 124 -> 118 0  Position: -3.08865  1.00493 -1.54941
IMG_3256.JPG 117 -> 119 0  Position: -1.97813 0.847976 -2.73495
IMG_3257.JPG 123 -> 120 0  Position: -3.12077 0.521545 -1.32637
IMG_3258.JPG 118 -> 121 0  Position:  -3.32749  0.122532 -0.715027
IMG_3259.JPG 116 -> 122 0  Position:  -3.20416 -0.348489  -1.07578
IMG_3260.JPG 126 -> 123 0  Position:  -2.43969 -0.804677  -2.25519
IMG_3261.JPG 128 -> 124 0  Position:   -1.3174 -0.434298  -3.15965
IMG_3262.JPG 122 -> 125 0  Position: -0.849994  -1.30582  -3.07265
IMG_3263.JPG 129 -> 126 0  Position: -0.885687  0.671677  -3.29061
IMG_3264.JPG 127 -> 127 0  Position:  0.365474 -0.413718  -3.43778
IMG_3265.JPG 132 -> 128 0  Position:  1.23839  1.08666 -3.22921
IMG_3266.JPG 131 -> 129 0  Position:    2.83986 0.00887473   -2.47471
IMG_3267.JPG 130 -> 130 0  Position:   3.03524 -0.799983  -2.07262
IMG_3268.JPG 134 -> 131 0  Position:  3.79573 -0.14781 0.358945
IMG_3269.JPG 140 -> 132 0  Position:   3.51369 -0.907138   1.35246
IMG_3270.JPG 135 -> 133 0  Position:  1.62936 -1.42592  3.18173
IMG_3271.JPG 136 -> 134 0  Position:  -1.34075 -0.954256   3.42907
IMG_3272.JPG 133 -> 135 0  Position:  -1.78598 0.0683145   3.36795
IMG_3273.JPG 137 -> 136 0  Position:  -3.12549 -0.145547   2.23101
IMG_3274.JPG 138 -> 137 0  Position:  -2.50854 -0.590182   2.52826
IMG_3275.JPG 139 -> 138 0  Position: -2.44068 0.406285  2.63433
IMG_3276.JPG 141 -> 139 0  Position: -2.61895 -0.47884  1.89863
IMG_3277.JPG 150 -> 140 0  Position: -0.918677 -0.535889   3.07764
IMG_3278.JPG 148 -> 141 0  Position:   1.09713 -0.501156   3.03876
IMG_3279.JPG 112 -> 142 0  Position:  -2.15122 -0.520614   1.77558
IMG_3280.JPG 151 -> 143 0  Position:  -2.25433 0.0936165    1.7139
IMG_3281.JPG 142 -> 144 0  Position:   -2.8706 0.0356229  -0.35274
IMG_3282.JPG 143 -> 145 0  Position:  -2.4946 0.166674 -1.52057
IMG_3283.JPG 149 -> 146 0  Position:    -1.422 -0.208194   1.71032
IMG_3284.JPG 145 -> 147 0  Position:   -2.16418 -0.0856412   0.663179
IMG_3285.JPG 144 -> 148 0  Position:  -2.25847 0.0249366 -0.395638
IMG_3286.JPG 147 -> 149 0  Position: -1.98711  0.46558 0.968542
IMG_3287.JPG 160 -> 150 0  Position: -0.780502  0.538116   2.01135
IMG_3288.JPG 153 -> 151 0  Position:  0.69962 0.137278  2.09652
IMG_3289.JPG 156 -> 152 0  Position: -0.393177 -0.466306   1.63222
IMG_3290.JPG 152 -> 153 0  Position:  -1.41748 -0.245603   1.01653
IMG_3291.JPG 146 -> 154 0  Position:    -1.7934 -0.0421622   0.214654
IMG_3292.JPG 159 -> 155 0  Position:  0.986653 -0.228469   1.42069
IMG_3293.JPG 155 -> 156 0  Position:   1.50458 -0.180417  0.925178
IMG_3294.JPG 154 -> 157 0  Position: 0.627464 0.616862  1.51369
IMG_3295.JPG 158 -> 158 0  Position: -0.524347 -0.216598   1.63262
IMG_3296.JPG 157 -> 159 0  Position:   1.11296 -0.280754   1.37764
IMG_3297.JPG 162 -> 160 0  Position: -0.786466 -0.469594   1.51335
IMG_3298.JPG 163 -> 161 0  Position:  -1.56089 -0.298944   0.81789
IMG_3299.JPG 171 -> 162 0  Position: -0.801103 -0.672253   1.43576
IMG_3300.JPG 167 -> 163 0  Position:    1.4354 -0.476129  0.947525
IMG_3301.JPG 165 -> 164 0  Position:  1.80069 -0.14865 0.143556
IMG_3302.JPG 173 -> 165 0  Position:   1.62079 -0.235785   1.60708
IMG_3303.JPG 164 -> 166 0  Position:  0.601838 -0.402867   2.13718
IMG_3304.JPG 168 -> 167 0  Position: -0.729843 -0.336501   2.08777
IMG_3305.JPG 166 -> 168 0  Position:  -1.28517 -0.287689   1.80982
IMG_3306.JPG 172 -> 169 0  Position:    2.31573 -0.0237463  -0.472049
IMG_3307.JPG 170 -> 170 0  Position:   2.03769 0.0716046  -1.23635
IMG_3308.JPG 169 -> 171 0  Position:   2.10588 -0.304728  -1.05297
IMG_3309.JPG 174 -> 172 0  Position:  2.09343 0.255348  0.97111
IMG_3310.JPG 177 -> 173 0  Position: 0.757424 0.386501  2.11794
IMG_3311.JPG 179 -> 174 0  Position:   1.64078 -0.313315   1.89945
IMG_3312.JPG 184 -> 175 0  Position:  2.34001 -0.54194   2.5011
IMG_3313.JPG 176 -> 176 0  Position:   0.73436 -0.774305   3.25724
IMG_3314.JPG 178 -> 177 0  Position: 0.526748 -0.65837   3.5036
IMG_3315.JPG 180 -> 178 0  Position:   1.5262 0.718477  3.89047
IMG_3316.JPG 185 -> 179 0  Position: 3.78793 0.12256 2.04468
IMG_3317.JPG 181 -> 180 0  Position:  3.68765 -1.03554  1.99838
IMG_3319.JPG 175 -> 181 0  Position:   1.22199 -0.547814   4.36642
IMG_3320.JPG 187 -> 182 0  Position:   2.65601 -0.501892    3.7226
IMG_3321.JPG 183 -> 183 0  Position:   3.79009 -0.295224   2.64246
IMG_3322.JPG 182 -> 184 0  Position: 1.56691 1.42827 4.07817
IMG_3323.JPG 186 -> 185 0  Position:  1.32505 0.706109  4.32447
Num Point3D 68636
Assertion 'info.ExposureBiasValue == 0' failed!
  File: /home/bart/projects/ADOP/src/apps/colmap2adop.cpp:44
  Function: std::vector<float, std::allocator<float> > ExposureValuesFromImages(std::vector<std::__cxx11::basic_string<char> >, std::string)
Aborted (core dumped)
```

Any help would be appreciated.

Thank you,

Bart

**Comment by trzy** [comment]
Looking at colmap2adop.cpp I see that this is a property of the JPEG header. Unfortunately, I can't go back and recapture the scene. Perhaps I can force this value to 0 in colmap2adop?

**Comment by darglein** [comment]
Should be fixed with 1b20452c9eeaaa5d2d1e5d0d03445da66a5c0194
